Sources/StreamChat/Repositories/OfflineRequestsRepository.swift

```diff
@@ -101,8 +107,36 @@ class OfflineRequestsRepository {
         data: Data,
         completion: @escaping () -> Void
     ) {
-        // TODO:
-        completion()
+        func decodeTo<T: Decodable>(_ type: T.Type) -> T? { try? JSONDecoder.stream.decode(T.self, from: data) }
+
+        switch endpoint.path {
+        case .createChannel:
+            guard let payload = decodeTo(ChannelPayload.self) else {
```

Comment:
Just a summary of this case - If the decoding fails we delete the queued request and the change is not applied. In which scenarios can this happen?

Reply:
It should actually not happen at all. Because we know the path of the request, and the return type is always the same (or at least it should), it should never happen. If the response were to come malformed, that could be this one case.